packages/system/src/Item/ListView/ListView.tsx

```diff
@@ -27,64 +31,33 @@ const listQueryFn = async ({
   totalCount: number;
 }> => {
   // TODO: Need to add a `sortByKey` to the Column type
-  const key = sortBy.key === 'name' ? 'NAME' : 'CODE';
+  const key =
+    sortBy.key === 'name' ? ItemSortFieldInput.Name : ItemSortFieldInput.Code;
 
-  const { items } = await request(
-    Environment.API_URL,
-    gql`
-      query items(
-        $first: Int
-        $offset: Int
-        $key: ItemSortFieldInput!
-        $desc: Boolean
-      ) {
-        items(
-          page: { first: $first, offset: $offset }
-          sort: { key: $key, desc: $desc }
-        ) {
-          ... on ItemConnector {
-            totalCount
-            nodes {
-              id
-              code
-              availableBatches {
-                ... on StockLineConnector {
-                  nodes {
-                    availableNumberOfPacks
-                    batch
-                    costPricePerPack
-                    expiryDate
-                    id
-                    itemId
-                    packSize
-                    sellPricePerPack
-                    storeId
-                    totalNumberOfPacks
-                  }
-                }
-                ... on ConnectorError {
-                  __typename
-                  error {
-                    description
-                  }
-                }
-              }
-              isVisible
-              name
-            }
-          }
-        }
-      }
-    `,
-    {
-      first,
-      offset,
-      key,
-      desc: sortBy.isDesc,
-    }
-  );
+  const { items } = await api.items({
+    first,
+    offset,
+    key,
+    desc: sortBy.isDesc,
+  });
+
+  if (items.__typename === 'ItemConnector') {
+    const itemRows: Item[] = items.nodes.map(item => ({
+      ...item,
+      availableQuantity: 0,
+      unit: '',
+      availableBatches:
```

Comment:
Nice! 

There's another use of the item list unfortunately - the add item modal has another version of the query and assigning an item throws an error. My solution is to export this query and use it in that modal.. but also requires updating of the `useListData` hook to remove the page size, otherwise only 20 items show!